src/Mapping/ExtensionMetadataFactory.php

```diff
@@ -66,8 +69,22 @@ class ExtensionMetadataFactory
      */
     private $cacheItemPool;
 
+    /**
+     * @param Reader|AttributeReader|object $annotationReader
+     */
     public function __construct(ObjectManager $objectManager, string $extensionNamespace, object $annotationReader, ?CacheItemPoolInterface $cacheItemPool = null)
     {
+        if (!$annotationReader instanceof Reader && !$annotationReader instanceof AttributeReader) {
+            trigger_deprecation(
+                'gedmo/doctrine-extensions',
+                '3.11',
+                'Providing an annotation reader which does not implement %s or is not an instance of %s to %s is deprecated.',
```

Comment:
We SHOULD tell the user the object type of `$annotationReader`.

Reply:
Usually, the exception messages regarding unsupported types deliver a message that includes the involved type. Check [this example](https://3v4l.org/Afjhc).

Reply:
> Providing an annotation reader which does not implement Doctrine\Common\Annotations\Reader or is not an instance of Gedmo\Mapping\Driver\AttributeReader to Gedmo\Mapping\ExtensionMetadataFactory is deprecated.

This is already a pretty lengthy message when you factor in the FQCN's of the three things involved.  If I'm changing the message to include the type that was passed in, for me I'd probably then remove the part about what the requirements are to keep the message from being too verbose:

> Providing My\NonCompliant\AnnotationReader as an annotation reader to Gedmo\Mapping\ExtensionMetadataFactory is deprecated and will cause a TypeError in 4.0.

Now the message is concise, but it's not really giving much info on how to fix it.  So to convey every bit of information, you end up with something like this:

> Providing My\NonCompliant\AnnotationReader as an annotation reader to Gedmo\Mapping\ExtensionMetadataFactory is deprecated and will cause a TypeError in 4.0.  As of 4.0, the annotation reader must implement Doctrine\Common\Annotations\Reader or be an instance of Gedmo\Mapping\Driver\AttributeReader.

Great, now we have all the information in the deprecation message.  But, that's a lot of information to consume for my tastes.

Now, to be fair, PHP core itself does specify the type in the deprecation message (see https://3v4l.org/YZmap#v8.2.0), but, I don't think PHP core has too many 300-character deprecation messages like the long version above is and I'd make the same case for information overload there as well.

Reply:
This isn't a blocker for me, just a suggestion. If you think there is no need to apply this, please omit the comment.

Reply:
I think I'm going to leave this alone for now, but I do think it's better to have explained why I like it as is instead of just ignoring the suggestion or accepting it without question.

Reply:
Thank you! :+1: